src/inmanta/agent/handler.py

```diff
@@ -574,6 +563,61 @@ def do_reload(self, ctx: HandlerContext, resource: resources.Resource) -> None:
         :param resource: The resource to reload.
         """
 
+    def pre(self, ctx: HandlerContext, resource: resources.Resource) -> None:
+        """
+        Method executed before a handler operation (Facts, dryrun, real deployment, ...) is executed. Override this method
+        to run before an operation.
+
+        :param ctx: Context object to report changes and logs to the agent and server.
+        :param resource: The resource to query facts for.
+        """
+
+    def post(self, ctx: HandlerContext, resource: resources.Resource) -> None:
+        """
+        Method executed after an operation. Override this method to run after an operation.
+
+        :param ctx: Context object to report changes and logs to the agent and server.
+        :param resource: The resource to query facts for.
+        """
+
+    def facts(self, ctx: HandlerContext, resource: resources.Resource) -> Dict[str, object]:
```

Comment:
I think this should be in the utility methods section. Can you check for the other methods you have lifted in the past few days? I think `pre` and `post` could also be considered utility methods if we define the interface methods as the "agent-handler API interface".